GameTest.java

```diff
@@ -0,0 +1,215 @@
+	}
+
+	// Down
+	
+	@Test
+	public void PacmanDownWall(){
+		// arrange
+		this.game.board = new Board(new String[]{"PW", "WG"}).getBoard();
+		Location l = new Location(0, 0);
+		
+		// action
+		game.movePacman(l, Game.Directions.Down);
+		
+		// assert
+		assertEquals(Board.Creatures.Pacman , this.game.board[l.getX()] [l.getY()]);
+		assertEquals(Board.Creatures.Wall , this.game.board[l.getX() + 1] [l.getY()]);
+	}
+	
+	// - Ghosts -
+	// Right
+	
+	@Test
+	public void GhostRightWall(){
+		// arrange
+		this.game.board = new Board(new String[]{"GW"}).getBoard();
+		Location l = new Location(0, 0);
+		
+		// action
+		game.moveGhost1(l, Game.Directions.Right);
+		
+		// assert
+		assertEquals(Board.Creatures.Ghost1, this.game.board[l.getX()] [l.getY()]);
+		assertEquals(Board.Creatures.Wall, this.game.board[l.getX()] [l.getY() + 1]);
+	}
+
+	@Test
+	public void GhostRightPoint_FirstTurn(){
+		// arrange
+		this.game.board = new Board(new String[]{"G-"}).getBoard();
+		Location l = new Location(0, 0);
+		
+		// action
+		game.moveGhost1(l, Game.Directions.Right);
+		
+		// assert
+		assertEquals(Board.Creatures.Ghost1, this.game.board[l.getX()] [l.getY() + 1]);
+		assertEquals(Board.Creatures.Null, this.game.board[l.getX()] [l.getY()]);
+	}
+
+	@Test
+	public void GhostRightPoint_SecondTurn(){
+		// arrange
+		this.game.board = new Board(new String[]{"G-."}).getBoard();
+		Location l = new Location(0, 0);
+		
+		// action
+		game.moveGhost1(l, Game.Directions.Right);
+		game.moveGhost1(new Location(0, 1), Game.Directions.Right);
+		
+		// assert
+		assertEquals(Board.Creatures.Ghost1, this.game.board[l.getX()] [l.getY() + 2]);
+		assertEquals(Board.Creatures.Point, this.game.board[l.getX()] [l.getY() + 1]);
+		assertEquals(Board.Creatures.Null, this.game.board[l.getX()] [l.getY()]);
+	}
+
+	@Test
+	public void Sorter(){
+		// arrange
+		this.game.board = new Board(new String[]{".G."}).getBoard();
+		this.game.savePoints = new Location[]{
+				null, new Location(0, 0), null, new Location(0, 2) };
+		
+		// action
+		this.game.sortNullsOnArray();
+		
+		// assert
+		assertEquals(null, this.game.savePoints[2]);
+		assertEquals(null, this.game.savePoints[3]);
+	}
+
+	@Test
+	public void Relocator(){
+		// arrange
+		this.game.board = new Board(new String[]{".G."}).getBoard();
+		this.game.savePoints = new Location[]{
+				new Location(0, 0), new Location(0, 2), null, null };
+		this.game.pointsSavedNum = 2; // 2 points to relocate
+		
+		// action
+		game.relocatePoints();
+		
+		// assert
+		assertEquals(Board.Creatures.Point, this.game.board[0][0]);
+		assertEquals(Board.Creatures.Point, this.game.board[0][2]);
+	}
+  
+	@Test
+	public void GhostRightPacman(){
+		// arrange
+		this.game.board = new Board(new String[]{"GP"}).getBoard();
+		Location l = new Location(0, 0);
+
+		// action
+		game.moveGhost1(l, Game.Directions.Right);
+		
+		// assert
+		assertEquals(Board.Creatures.Ghost1, this.game.board[l.getX()][l.getY() + 1]);
```

Comment:
Does ghosts "eat" Pacman?
Maybe you want to "signal" that the game is over? 
It's not a regular move, so you must treat it in different way